repo.go

```diff
@@ -167,6 +167,40 @@ func (r *Repo) RootVersion() (int, error) {
 	return root.Version, nil
 }
 
+func (r *Repo) GetThreshold(keyRole string) (int, error) {
+	root, err := r.root()
+	if err != nil {
+		return -1, err
+	}
+	role, ok := root.Roles[keyRole]
+	if !ok {
+		return -1, ErrInvalidRole{keyRole}
+	}
+
+	return role.Threshold, nil
+}
+
+func (r *Repo) SetThreshold(keyRole string, t int) error {
+	root, err := r.root()
+	if err != nil {
+		return err
+	}
+	role, ok := root.Roles[keyRole]
+	if !ok {
+		return ErrInvalidRole{keyRole}
+	}
+	if role.Threshold == t {
+		// Change was a no-op.
+		return nil
+	}
+	role.Threshold = t
+	if _, ok := r.versionUpdated["root.json"]; !ok {
```

Comment:
Hmm, this may have to change for delegations. If the threshold of a delegated target is updated, it's the parent target that will need a new version.

Reply:
Hmmmmm that's a good point. I took a look at https://github.com/theupdateframework/go-tuf/pull/130 and I don't even think that properly handles incrementing the parent roles (only targets, maybe not an intermediate).

should i error out if it's not a top-level role right now, or since delegations aren't even supported hold off?

Reply:
Yeah, maybe error out for now. Eventually this'll need some kind of knowledge of who the parent of a role is, but that can be left to the delegations pr.